draft-ietf-add-split-horizon-authority.xml

```diff
@@ -179,158 +185,163 @@
+          specification.</li>
+        <li>Least privilege: Local resolvers do not hold any
+          secrets that could weaken the security of the public zone if
+          compromised.</li>
+        <li>Local zone confidentiality: The specification does not leak
+          local network subdomains to anyone outside of the network.</li>
+        <li>Flexibility: The specification can represent and authorize
+          any reasonable Split DNS zone structure.</li>
+        <li>DNSSEC Compatibility: The specification supports DNSSEC-based
+          object security for local zone contents.</li>
+      </ul>
     </section>
-    <!-- Learning -->
-
     <section anchor="establishing">
       <name>Establishing Local DNS Authority</name>
       <t>To establish its authority over some DNS zone, a
       participating network <bcp14>MUST</bcp14> offer one or more
       encrypted resolvers via DNR <xref target="I-D.ietf-add-dnr"/>,
       DDR <xref target="I-D.ietf-add-dnr"/>, or an equivalent mechanism
-      (see <xref target="vpn"/>).  If the encrypted resolver is
-      identified by name (e.g., DNR), at least one of these resolvers'
-      Authentication Domain Names (ADNs) <bcp14>MUST</bcp14> appear in
-      an NS record for that zone.  If the encrypted resolver is
-      identified by IP address (e.g., DDR), then the resolver MUST present a
-      certificate that can be validated by the client, containing at least
-      one subjectAltName entry that matches an NS record for that zone.
-      This arrangement establishes this resolver's authority over the zone.</t>
+      (see <xref target="vpn"/>).</t>
+      <t>To establish local authority, the network MUST convey one or more
+      "Authorization Claims" to the client.  An "Authorization Claim" is an
+      abstract structure comprising:</t>
+      <ul>
+        <li>An Authentication Domain Name (ADN) of a local encrypted resolver.</li>
+        <li>The DNS name of the authorizing parent zone.</li>
+        <li>A list of subdomains of this parent zone that are claimed by
+            the named local resolver (potentially including the entire parent
+            zone).</li>
+        <li>An NSEC3 Hash Algorithm.</li>
+        <li>A high-entropy salt.</li>
+      </ul>
+      <t>If the local encrypted resolver is identified by name (e.g., DNR), that
+      identifying name MUST be the one used in any corresponding Authorization
+      Claim.  Otherwise (e.g., DDR using IP addresses), the resolver MUST
+      present a validatable certificate containing a subjectAltName that
+      matches the Authorization Claim.</t>
+      <t>To establish its authority, the network MUST provide each Authorization
+      Claim to the parent zone operator.  If the contents are approved, the
+      parent zone operator computes a "Verification Token" according to the
+      following procedure based on <relref section="5" target="RFC5155"/>:</t>
```

Comment:
I see RFC5155 only uses SHA-1 with Hashed Message Authentication Code (HMAC) and do we have add new hash algorithms in the "DNSSEC NSEC3 Hash Algorithms" registry ?

Reply:
Looks good to me.